builder/azure/arm/template_factory.go

```diff
@@ -1,24 +1,46 @@
 package arm
 
 import (
+	"crypto/rsa"
 	"encoding/json"
+	"errors"
+	"fmt"
 
 	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-11-01/compute"
 	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-02-01/resources"
-
-	"fmt"
-
 	"github.com/hashicorp/packer-plugin-azure/builder/azure/common/constants"
 	"github.com/hashicorp/packer-plugin-azure/builder/azure/common/template"
+	"golang.org/x/crypto/ssh"
 )
 
 type templateFactoryFunc func(*Config) (*resources.Deployment, error)
 
-func GetKeyVaultDeployment(config *Config) (*resources.Deployment, error) {
+func GetCommunicatorSpecificKeyVaultDeployment(config *Config) (*resources.Deployment, error) {
```

Comment:
the ssh changes intermixed are making it a bit harder to review the custom script change independently. Is it possible to split?

Reply:
+1 would appreciate if we could remove the SSH changes, targeting this branch against your first PR instead of main that should resolve this, and when your first PR is merged GitHub will auto update this PR to be on top of the new main

Reply:
Forgot to ping @matt-richardson

Reply:
I had a quick look, but because it's from a repo fork, it wont let me change the target branch 
![image](https://user-images.githubusercontent.com/373389/206778337-3c244694-e5a6-473b-b689-8bdafe574415.png)

I'd have to close and re-open the PR to do that.

If we're able to get the SSH PR through first, then that would work, wouldn't it? 
Or is there something blocking that from happening?

Reply:
That'll be OK, it makes it a bit more confusing but it doesn't necessarily block that PR, I am still working on validating that, hoping to find the time to keep working on that today or next week, however I do notice now that that PR has a merge conflict after a recent other PR merge, can you take a look @matt-richardson